sockeye/layers.py

```diff
@@ -46,11 +48,1542 @@ def activation(data: mx.sym.Symbol, act_type: str) -> mx.sym.Symbol:
+    """
+    Position-wise feed-forward network with activation.
+    """
+
+    def __init__(self,
+                 num_hidden: int,
+                 dropout: float,
+                 no_bias: bool,
+                 prefix: str) -> None:
+        self.num_hidden = num_hidden
+        self.dropout = dropout
+        self.no_bias = no_bias
+        self.prefix = prefix
+        self.w_i2h = mx.sym.Variable('%si2h_weight' % prefix)
+        self.b_i2h = None if no_bias else mx.sym.Variable('%si2h_bias' % prefix)
+
+    def _fc(self, data: mx.sym.Symbol):
+        bias = None if self.no_bias else self.b_i2h
+        h = mx.sym.FullyConnected(data=data, num_hidden=self.num_hidden, weight=self.w_i2h, bias=bias,
+                                  no_bias=self.no_bias, flatten=False)
+        if self.dropout > 0.0:
+            h = mx.sym.Dropout(h, p=self.dropout)
+        return h
+
+    def __call__(self,
+                 data: mx.sym.Symbol,
+                 lengths: Optional[mx.sym.Symbol] = None,
+                 max_length: Optional[int] = None) -> mx.sym.Symbol:
+        """
+        Position-wise feed-forward network with activation.
+
+        :param x: Symbol of shape (batch_size, seq_len, num_hidden)
+        :return: Symbol of shape (batch_size, seq_len, num_hidden)
+        """
+        return self._fc(data)
+
+    def get_num_hidden(self) -> int:
+        return self.num_hidden
+
+
+class LinearLayerConfig(LayerConfig):
+
+    def __init__(self, num_hidden: int, dropout: float, no_bias=False):
+        super().__init__()
+        self.num_hidden = num_hidden
+        self.dropout = dropout
+        self.no_bias = no_bias
+
+    def create_encoder_layer(self, input_num_hidden: int, prefix: str) -> EncoderLayer:
+        return StatelessBlockLayer(LinearBlock(self.num_hidden, self.dropout, no_bias=self.no_bias,
+                                               prefix=prefix + "linear_"))
+
+    def create_decoder_layer(self, input_num_hidden: int, prefix: str) -> DecoderLayer:
+        return StatelessBlockLayer(LinearBlock(self.num_hidden, self.dropout, no_bias=self.no_bias,
+                                               prefix=prefix + "linear_"))
+
+
+class ActivationBlock(StatelessBlock):
+    """
+    Position-wise feed-forward network with activation.
+    """
+
+    def __init__(self,
+                 act_type: str,
+                 num_hidden: int) -> None:
+        self.act_type = act_type
+        self.num_hidden = num_hidden
+
+    def __call__(self,
+                 data: mx.sym.Symbol,
+                 lengths: Optional[mx.sym.Symbol] = None,
+                 max_length: Optional[int] = None) -> mx.sym.Symbol:
+        """
+        Position-wise feed-forward network with activation.
+
+        :param x: Symbol of shape (batch_size, seq_len, num_hidden)
+        :return: Symbol of shape (batch_size, seq_len, num_hidden)
+        """
+        data = mx.sym.Activation(data, act_type=self.act_type)
```

Comment:
could this use `activation()` as well (supporting our additional activation functions, swish etc)?

Reply:
makes sense, done!